scripts/seed_users.js

```diff
@@ -0,0 +1,530 @@
+
+// Initiate Server
+function initServer() {
+    // Import sugarizer server
+    require('../sugarizer');
+}
+
+// Initiate seeding to DB
+function initSeed() {
+    // Create Master Admin
+    console.log('Creating Temporary Admin...');
+
+    request({
+        headers: {
+            "content-type": "application/json",
+        },
+        json: true,
+        method: 'POST',
+        uri: common.getAPIUrl() + 'auth/signup',
+        body: {
+            "user" : masterAdmin
+        }
+    }, function(error, response) {
+        if (response.statusCode == 200) {
+            console.log('Temporary Admin created');
+            request({
+                headers: {
+                    "content-type": "application/json",
+                },
+                json: true,
+                method: 'POST',
+                uri: common.getAPIUrl() + 'auth/login',
+                body: {
+                    "user" : masterAdmin
+                }
+            }, function(error, response, body) {
+                if (response.statusCode == 200) {
+                    masterAdmin = body;
+
+                    console.log('Logged into Temporary Admin');
+
+                    console.log('Inserting users...');
+                    var usersProcessed = 0;
+                    // Insert all users
+                    for (var i=0; i < Users.length; i++) {
+                        insertUser(i).then(function() {
+                            usersProcessed++;
+                            if (usersProcessed == Users.length) initClassroomAssignment();
+                        }).catch(function(err) {
+                            usersProcessed++;
+                            console.log(err);
+                            if (usersProcessed == Users.length) initClassroomAssignment();
+                        });
+                    }
+                } else {
+                    console.log('Error logging into Temporary Admin');
+                    process.exit(-1);
+                }
+            });
+        } else {
+            console.log('Error creating Temporary Admin');
+            process.exit(-1);
+        }
+    });
+}
+
+console.log("Reading and validating file");
+var dataIndex = 0;
+// Read the CSV file
+fs.createReadStream(filename)
+    .on('error', function(err) {
+        console.log(err);
+        // ToDo: Check safely terminate script
+        throw err;
+    })
+    .pipe(csv())
+    .on('data', function(row) {
+        dataIndex++;
+        var validRow = validateUserRow(row, dataIndex);
+        if (validRow) Users.push(new User(validRow.name, validRow.type, validRow.language, validRow.color, validRow.password, validRow.classroom));
+    })
+    .on('end', function() {
+        console.log('Finished processing CSV file');
+        console.log('Starting Sugarizer-Server...');
+        if (Users.length == 0) {
+            console.log('Error: No users to insert');
+            process.exit(-1);
+        }
+        initServer();
```

Comment:
Hmmm. The script should consider that the server is already running. It's strange to be forced to stop the server to launch the script.

Reply:
I wrote the script assuming that the server is not running. I'll remove the `initServer` part.